cli/data.py

```diff
@@ -237,12 +234,22 @@ def update_case_based_icu_utilization_weights():
         json.dump(output, f, indent=2, sort_keys=True)
 
 
-def build_field_dataset_source(feature_definition_config):
+def build_field_datasets(
+    feature_definition_config: combined_datasets.FeatureDataSourceMap,
```

Comment:
How about doing the filtering here? instead of in a separate method.

```python
def build_field_datasets(
    feature_definition_config: combined_datasets.FeatureDataSourceMap,
    state=None,
    fips=None
):

    def _load_dataset(data_source_cls):
        dataset = data_source_cls.local().multi_region_dataset()
        if state or fips:
            dataset = dataset.get_subset(state=state, fips=fips)
        return dataset
    
    feature_definition = {
        field_name: list(reversed(list(_load_dataset(cls) for cls in classes)))
        for field_name, classes in feature_definition_config.items()
        if classes
    }
    return feature_definition    
```

Reply:
I've gone back and forth on this but hadn't tried an internal function. Done.